firebase-firestore/src/commonTest/kotlin/dev/gitlive/firebase/firestore/FieldValueTests.kt

```diff
@@ -0,0 +1,24 @@
+package dev.gitlive.firebase.firestore
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+class FieldValueTests {
+    @Test
+    fun equalityChecks() = runTest {
+        assertEquals(FieldValue.delete, FieldValue.delete)
+        assertEquals(FieldValue.delete(), FieldValue.delete())
+        assertEquals(FieldValue.serverTimestamp(), FieldValue.serverTimestamp())
+        assertNotEquals(FieldValue.delete, FieldValue.serverTimestamp())
+
+        // Note: arrayUnion and arrayRemove can't be checked due to vararg to array conversion
+//        assertEquals(FieldValue.arrayUnion(1, 2, 3), FieldValue.arrayUnion(1, 2, 3))
```

Comment:
any suggestions to have equality working as expected?

Reply:
`assertContentEquals`?

Reply:
Can you make Set out of Array?